### ISSUE TITLE
sfcgal dependency for postgis failed to build on 10.10.5

### ISSUE THREAD

**Issue by iBeloved** [Oslandia/SFCGAL#165]
Hi.
I was deploying PostGIS via home-brew when I struct a problem...
Then i run brew install postgis ,and I got the following (sfcgal part only):

==> Installing postgis dependency: sfcgal
==> Downloading https://github.com/Oslandia/SFCGAL/archive/v1.3.5.tar.gz
Already downloaded: /Users/......./Library/Caches/Homebrew/sfcgal-1.3.5.tar.gz
==> cmake . -DCMAKE_C_FLAGS_RELEASE=-DNDEBUG -DCMAKE_CXX_FLAGS_RELEASE=-DNDEBUG -DCMAKE_INSTALL_PREFIX=/usr/local/Cellar/sfcgal/1.3.5 -DCMAKE_BUILD_TYPE=Release -DCMAKE_FIND_FRAMEWORK=LAST -DCMAKE_VERBOSE_MAKEFILE=ON -Wno-dev
==> make install
Last 15 lines from /Users/......./Library/Logs/Homebrew/sfcgal/02.make:
      ...
  "typeinfo for boost::archive::detail::basic_pointer_oserializer", referenced from:
      typeinfo for boost::archive::detail::pointer_oserializer<boost::archive::binary_oarchive, SFCGAL::Point> in Serialization.cpp.o
      typeinfo for boost::archive::detail::pointer_oserializer<boost::archive::binary_oarchive, SFCGAL::LineString> in Serialization.cpp.o
      typeinfo for boost::archive::detail::pointer_oserializer<boost::archive::binary_oarchive, SFCGAL::Triangle> in Serialization.cpp.o
      typeinfo for boost::archive::detail::pointer_oserializer<boost::archive::binary_oarchive, SFCGAL::Polygon> in Serialization.cpp.o
      typeinfo for boost::archive::detail::pointer_oserializer<boost::archive::binary_oarchive, SFCGAL::TriangulatedSurface> in Serialization.cpp.o
      typeinfo for boost::archive::detail::pointer_oserializer<boost::archive::binary_oarchive, SFCGAL::PolyhedralSurface> in Serialization.cpp.o
      typeinfo for boost::archive::detail::pointer_oserializer<boost::archive::binary_oarchive, SFCGAL::Solid> in Serialization.cpp.o
      ...
ld: symbol(s) not found for architecture x86_64
clang: error: linker command failed with exit code 1 (use -v to see invocation)
make[2]: *** [src/libSFCGAL.1.3.5.dylib] Error 1
make[1]: *** [src/CMakeFiles/SFCGAL.dir/all] Error 2
make: *** [all] Error 2

Anyone struck this problem? seems to have a boost component issue which is the latest.
Any help would be really appreciated.

**Comment by vmora** [comment]
@iBeloved thanks for the report.

Could you plz attach the full log from `make VERBOSE=1`

**Comment by iBeloved** [comment]
Here's everything from the previous command.

==> Downloading https://github.com/Oslandia/SFCGAL/archive/v1.3.5.tar.gz
Already downloaded: /Users/........./Library/Caches/Homebrew/sfcgal-1.3.5.tar.gz
==> cmake . -DCMAKE_C_FLAGS_RELEASE=-DNDEBUG -DCMAKE_CXX_FLAGS_RELEASE=-DNDEBUG -DCMAKE_INSTALL_PREFIX=/usr/local/Cellar/sfcgal/1.3.5 -DCMAKE_BUILD_TYPE=Release -DCMAKE_FIND_FRAMEWORK=LAST -DCMAKE_VERBOSE_MAKEFILE=ON -Wno-dev
==> make install
Last 15 lines from /Users/....../Library/Logs/Homebrew/sfcgal/02.make:
      ...
  "typeinfo for boost::archive::detail::basic_pointer_oserializer", referenced from:
      typeinfo for boost::archive::detail::pointer_oserializer<boost::archive::binary_oarchive, SFCGAL::Point> in Serialization.cpp.o
      typeinfo for boost::archive::detail::pointer_oserializer<boost::archive::binary_oarchive, SFCGAL::LineString> in Serialization.cpp.o
      typeinfo for boost::archive::detail::pointer_oserializer<boost::archive::binary_oarchive, SFCGAL::Triangle> in Serialization.cpp.o
      typeinfo for boost::archive::detail::pointer_oserializer<boost::archive::binary_oarchive, SFCGAL::Polygon> in Serialization.cpp.o
      typeinfo for boost::archive::detail::pointer_oserializer<boost::archive::binary_oarchive, SFCGAL::TriangulatedSurface> in Serialization.cpp.o
      typeinfo for boost::archive::detail::pointer_oserializer<boost::archive::binary_oarchive, SFCGAL::PolyhedralSurface> in Serialization.cpp.o
      typeinfo for boost::archive::detail::pointer_oserializer<boost::archive::binary_oarchive, SFCGAL::Solid> in Serialization.cpp.o
      ...
ld: symbol(s) not found for architecture x86_64
clang: error: linker command failed with exit code 1 (use -v to see invocation)
make[2]: *** [src/libSFCGAL.1.3.5.dylib] Error 1
make[1]: *** [src/CMakeFiles/SFCGAL.dir/all] Error 2
make: *** [all] Error 2

Do not report this issue to Homebrew/brew or Homebrew/core!

Log output above ---------------
Blow is on terminal response ---

Error: You are using macOS 10.10.
We (and Apple) do not provide support for this old version.
You may encounter build failures or other breakages.
Please create pull-requests instead of filing issues.
Error: You are using macOS 10.10.
We (and Apple) do not provide support for this old version.
You may encounter build failures or other breakages.
Please create pull-requests instead of filing issues.

**Comment by iBeloved** [comment]
Found the log files at: 
~/Library/Logs/Homebrew/sfcgal
00.options.out
01.cmake
01.cmake.cc
02.make
02.make.cc
CMakeCache.txt

Searching each file only the 02.make file has a reference to `VERBOSE=1`, not as you described `make VERBOSE=1`

**Comment by iBeloved** [comment]
Here's extracted data from the `02.make` file.
Complete from the instance of `VERBOSE=1`

[verbose_extract.txt](https://github.com/Oslandia/SFCGAL/files/1968352/verbose_extract.txt)

**Comment by vmora** [comment]
Apparently the linker instructions are in CMakeFiles/SFCGAL.dir/link.txt under a temporary dir (/tmp/sfcgal-20180502-45250-1gqybo/SFCGAL-1.3.5/src in you log). I don't know if you can have access to the directory after the build fails, but if it's the case, I'd like to see the link options (since obviously something is missing).

Also, could you tell wich cmake version is used ?

@sloriot didn't you metionned to me something about new linking strategies/variables in cmake that are used by boost.

**Comment by iBeloved** [comment]
I’ll follow up on the file location a bit later.

The cmake version is 3.11.1

You may want to have a look at the uploaded txt file I scraped that I attached to the question yesterday. It does show references to boost in that file.

**Comment by sloriot** [comment]
@vmora serialization is not required by CGAL but SFCGAL. Maybe a missing dependency.

**Comment by mhugo** [comment]
Could you try with https://github.com/Oslandia/SFCGAL/releases/tag/v1.3.6 and tell us ?

**Comment by ascherkus** [comment]
I'm still encountering this issue with v.1.3.6 on Mac w/ CGAL v 4.14 and LLVM v10.0.1

```
$ uname -a
Darwin AndrewS-MBPT-TO.local 18.5.0 Darwin Kernel Version 18.5.0: Mon Mar 11 20:40:32 PDT 2019; root:xnu-4903.251.3~3/RELEASE_X86_64 x86_64

$ brew info boost
boost: stable 1.69.0 (bottled), HEAD
Collection of portable C++ source libraries
https://www.boost.org/
/src/homebrew/Cellar/boost/1.69.0 (13,717 files, 488.8MB)
  Poured from bottle on 2019-04-18 at 10:59:16
/src/homebrew/Cellar/boost/1.69.0_2 (13,717 files, 489.4MB) *
  Poured from bottle on 2019-04-22 at 17:16:11
From: https://github.com/Homebrew/homebrew-core/blob/master/Formula/boost.rb
==> Dependencies
Required: icu4c ✔
==> Options
--HEAD
	Install HEAD version
==> Analytics
install: 72,803 (30 days), 169,129 (90 days), 646,221 (365 days)
install_on_request: 24,440 (30 days), 55,329 (90 days), 197,422 (365 days)
build_error: 0 (30 days)

$ brew info cgal
cgal: stable 4.14 (bottled)
Computational Geometry Algorithm Library
https://www.cgal.org/
/src/homebrew/Cellar/cgal/4.14 (3,292 files, 36.4MB) *
  Poured from bottle on 2019-04-18 at 10:59:29
From: https://github.com/Homebrew/homebrew-core/blob/master/Formula/cgal.rb
==> Dependencies
Build: cmake ✔
Required: boost ✔, eigen ✔, gmp ✔, mpfr ✔
==> Analytics
install: 7,164 (30 days), 15,587 (90 days), 54,382 (365 days)
install_on_request: 938 (30 days), 1,704 (90 days), 6,938 (365 days)
build_error: 0 (30 days)

$ /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/c++ -v
Apple LLVM version 10.0.1 (clang-1001.0.46.4)
Target: x86_64-apple-darwin18.5.0
Thread model: posix
InstalledDir: /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin

$ mkdir build
$ cd build
$ cmake ../ -DCMAKE_C_FLAGS_RELEASE=-DNDEBUG -DCMAKE_CXX_FLAGS_RELEASE=-DNDEBUG -DCMAKE_BUILD_TYPE=Release -DCMAKE_FIND_FRAMEWORK=LAST -DCMAKE_VERBOSE_MAKEFILE=ON -Wno-dev
$ make

[ snip ]

cd /src/sfcgal/SFCGAL-1.3.6/build/src && /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/c++  -DBOOST_ALL_DYN_LINK -DBOOST_ALL_NO_LIB -DCGAL_USE_CORE=1 -DSFCGAL_BUILD_SHARED -DSFCGAL_EXPORTS -I/src/sfcgal/SFCGAL-1.3.6/build/include -I/src/sfcgal/SFCGAL-1.3.6/src/../CGAL_patches -isystem /src/homebrew/include  -DNDEBUG -isysroot /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.14.sdk -fPIC   -o CMakeFiles/SFCGAL.dir/Coordinate.cpp.o -c /src/sfcgal/SFCGAL-1.3.6/src/Coordinate.cpp
/src/sfcgal/SFCGAL-1.3.6/src/Coordinate.cpp:276:32: error: no matching function for call to 'round'
                               SFCGAL::round( v.exact() * _scaleFactor ),
                               ^~~~~~~~~~~~~
/src/sfcgal/SFCGAL-1.3.6/build/include/SFCGAL/numeric.h:50:15: note: candidate function not viable: no known conversion from 'typename enable_if<is_compatible_arithmetic_type<long, number<gmp_rational, et_on> >, detail::expression<detail::multiply_immediates, number<gmp_rational, et_on>, long> >::type' (aka 'boost::multiprecision::detail::expression<boost::multiprecision::detail::multiply_immediates, boost::multiprecision::number<boost::multiprecision::backends::gmp_rational, boost::multiprecision::et_on>, long, void, void>') to 'const double' for 1st argument
inline double round( const double& v )
              ^
/src/sfcgal/SFCGAL-1.3.6/build/include/SFCGAL/numeric.h:86:23: note: candidate function not viable: no known conversion from 'typename enable_if<is_compatible_arithmetic_type<long, number<gmp_rational, et_on> >, detail::expression<detail::multiply_immediates, number<gmp_rational, et_on>, long> >::type' (aka 'boost::multiprecision::detail::expression<boost::multiprecision::detail::multiply_immediates, boost::multiprecision::number<boost::multiprecision::backends::gmp_rational, boost::multiprecision::et_on>, long, void, void>') to 'const CGAL::Gmpq' for 1st argument
SFCGAL_API CGAL::Gmpz round( const CGAL::Gmpq& v ) ;
                      ^
1 error generated.
```

Wondering if it's issue here or with CGAL itself (possible https://github.com/CGAL/cgal/issues/3816 ?)

**Comment by lbartoletti** [comment]
Fixed in 1.3.7 and https://github.com/Oslandia/SFCGAL/pull/219 for CGAL 5
